Autoload/Lobby.gd

```diff
@@ -0,0 +1,69 @@
+extends Control
+
+var player = load("res://Scenes/Player.tscn")
+var ball = load("res://Scenes/Ball.tscn")
+var client_ball = load("res://Scenes/Client Ball.tscn")
+
+var username_text
+var port
+var number_of_players
+var ip
+
+func _ready():
+	get_tree().connect("network_peer_connected", self, "_player_connected")
+	get_tree().connect("network_peer_disconnected", self, "_player_disconnected")
+	get_tree().connect("connected_to_server", self, "_connected_to_server")
+	get_tree().connect("server_disconnected", self, "_server_disconnected")
+
+func _player_connected(id):
+	if id != get_tree().get_network_unique_id():
+		print("Player " + str(id) + " has connected")
+		instance_player(id)
+
+func _player_disconnected(id):
+	print("Player " + str(id) + " has disconnected")
+	
+	if Players.has_node(str(id)):
+		Players.get_node(str(id)).username_instance.queue_free()
+		Players.get_node(str(id)).queue_free()
+
+func _connected_to_server():
+	get_tree().change_scene("res://Scenes/World.tscn")
+	instance_player(get_tree().get_network_unique_id())
+	yield(get_tree().create_timer(0.1), "timeout")
+	Global.instance_client_ball(client_ball)
+
+func _server_disconnected():
+	for n in Players.get_children():
+		Players.remove_child(n)
```

Comment:
Is this even allowed/safe in this language?  Generally trying to change the set composition while iterating through it can lead to unintended behaviour.

Reply:
As far as I can tell, this seems pretty standard in this language. This works because it sends a constant array containing a reference to each child node. I think I get what you mean though, so I'll put it into a constant array first just for the sake of better clarity.